src/helpers/api.ts

```diff
@@ -203,6 +203,34 @@ export const fetchStudentData = async (
   return json;
 };
 
+export const fetchSeasonData = async (
+  accessToken: string | undefined,
+  filter: string | undefined
+) => {
+  if (!accessToken || accessToken === undefined) {
+    redirect();
+    return;
+  }
+  const res = await fetch(
+    filter ? url(`/registrations?${filter}`) : url("/registrations"),
+    {
+      next: { tags: ["AllStudents"] },
+      headers: {
+        Authorization: `Bearer ${accessToken}`,
+      },
+    }
+  );
+  const json = await res.json();
```

Comment:
```suggestion
  return await res.json();
```

Reply:
couldn't change it due to some error in promises